modules/serverless-logic-security-example-oauth-authentication.adoc

```diff
@@ -0,0 +1,102 @@
+// Module included in the following assemblies:
+// serverles-logic/serverless-logic-authentication-openapi-services
+
+:_mod-docs-content-type: REFERENCE
+[id="serverless-logic-security-example-oauth-authentication_{context}"]
+= Example of OAuth 2.0 authentication
+
+The following example shows how to secure an OpenAPI operation using the OAuth 2.0 `clientCredentials` flow. The OpenAPI specification defines the `sayHelloOauth2` operation, which uses the `oauth-example` security scheme. Unlike simpler authentication methods such as HTTP Basic or API keys, OAuth 2.0 authentication requires additional integration with the Quarkus OpenID Connect (OIDC) Client.
+
+.Example OpenAPI specification with OAuth 2.0
+[source,json]
+----
+{
+  "openapi": "3.1.0",
+  "info": {
+    "title": "Oauth2 Scheme Example",
+    "version": "1.0"
+  },
+  "paths": {
+    "/hello-with-oauth2": {
+      "get": {
+        "operationId": "sayHelloOauth2",
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "text/plain": {
+                "schema": {
+                  "type": "string"
+                }
+              }
+            }
+          }
+        },
+        "security": [
+          {
+            "oauth-example": []
+          }
+        ]
+      }
+    }
+  },
+  "components": {
+    "securitySchemes": {
+      "oauth-example": {
+        "type": "oauth2",
+        "flows": {
+          "clientCredentials": {
+            "authorizationUrl": "https://example.com/oauth",
+            "tokenUrl": "https://example.com/oauth/token",
+            "scopes": {}
+          }
+        }
+      }
+    }
+  }
+}
+----
+
+In this example, the `sayHelloOauth2` operation is protected by the `oauth-example` security scheme, which uses the `clientCredentials` flow for token-based authentication.
+
+[id="serverless-logic-security-oauth-support-oidc-client-filter-extention_{context}"]
+== OAuth 2.0 Support with the OIDC Client filter extension
+
+OAuth 2.0 token management is handled by a Quarkus `OidcClient`. To enable this integration, you must add the Quarkus OIDC Client Filter extension to your project as shown in the following examples:
+
+.Example of adding extension using Maven
+[source,xml]
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-oidc-client-filter</artifactId>
+</dependency>
+----
```

Comment:
```suggestion

<dependency>
  <groupId>io.quarkiverse.openapi.generator</groupId>
  <artifactId>quarkus-openapi-generator-oidc</artifactId>
  <version>2.9.0-lts</version>
</dependency>

----
```

Reply:
Hi, @kaldesai in the sugestion above :point_up: we are adding basically one new dependency.
with the version -> **2.9.0-tls** 

Pelease take a look if instead of adding the fixed value **2.9.0-lts** we can define a document variable, and use it in this configuration. In other parts of the documentation  IIRC, we have already done this to setup for instance some image  tags numbers :thinking:

Reply:
Yes, I have added the attribute for the same

Reply:
@kaldesai this change I don't see in the preview link https://92245--ocpdocs-pr.netlify.app/openshift-serverless/latest/serverless-logic/serverless-logic-managing-security/serverless-logic-authentication-openapi-services#serverless-logic-security-example-oauth-authentication_serverless-logic-authentication-openapi-services  :thinking: , maybe outdated?

Reply:
Not sure if was didn't explain well of the merged wasn't good, let me explain:

I the produced document we must see two maven dependencies:

<dependency>
  <groupId>io.quarkus</groupId>
  <artifactId>quarkus-oidc-client-filter</artifactId>
</dependency>

<dependency>
  <groupId>io.quarkiverse.openapi.generator</groupId>
  <artifactId>quarkus-openapi-generator-oidc</artifactId>
  <version>2.9.0-lts</version>
</dependency>

But I in the preview link I can see this:

![image](https://github.com/user-attachments/assets/29e58501-8a74-4fe9-bb2e-74593d567be8)

Reply:
@wmedvede so there should be two dependencies, correct? As follows?  

[source,xml]
----
<dependency>
<groupId>io.quarkus</groupId>
  <artifactId>quarkus-oidc-client-filter</artifactId>
</dependency>

<dependency>
  <groupId>io.quarkiverse.openapi.generator</groupId>
  <artifactId>quarkus-openapi-generator-oidc</artifactId>
  <version>{ServerlessLogicOauthDependencyVersion}</version>
</dependency>
----